OsmAnd-java/src/main/java/net/osmand/router/HHRoutePlanner.java

```diff
@@ -741,7 +743,15 @@ private HHRoutingContext<T> selectBestRoutingFiles(LatLon start, LatLon end, HHR
 			}
 		}
 		for (HHRouteRegionsGroup<T> g : groups) {
-			g.containsStartEnd = g.contains(start) && g.contains(end);
+			boolean cStart = g.contains(start);
```

Comment:
If group of HH route regions is selected only points from these group should be used. So it means there is a bug in route calculation but that check doesn't make sense